python/tests/test_reduce.py

```diff
@@ -0,0 +1,89 @@
+import pytest
+import torch
+from torch.testing import assert_close
+
+import triton
+import triton.language as tl
+
+
+def patch_kernel(template, to_replace):
+    kernel = triton.JITFunction(template.fn)
+    for key, value in to_replace.items():
+        kernel.src = kernel.src.replace(key, value)
+    return kernel
+
+
+@triton.jit
+def reduce1d_kernel(x_ptr, z_ptr, block: tl.constexpr):
+    x = tl.load(x_ptr + tl.arange(0, block))
+    tl.store(z_ptr, tl.OP(x, axis=0))
+
+
+@triton.jit
+def reduce2d_kernel(x_ptr, z_ptr, block_m: tl.constexpr, block_n: tl.constexpr):
+    range_m = tl.arange(0, block_m)
+    range_n = tl.arange(0, block_n)
+    x = tl.load(x_ptr + range_m[:, None] * block_n + range_n[None, :])
+    z = tl.OP(x, axis=AXIS)
+    if AXIS == 0:
+        tl.store(z_ptr + range_n, z)
+    else:
+        tl.store(z_ptr + range_m, z)
+
+
+reduce1d_configs = [
+    (op, shape) for op in ['sum', 'min', 'max'] for shape in [32, 64, 128, 512, 1024]
+]
+
+
+@pytest.mark.skip(reason="conversion from scalar to 0-d tensor not implemented")
+@pytest.mark.parametrize('op, shape', reduce1d_configs)
+def test_reduce1d(op, shape):
+    print('test_reduce1d, op = {}, shape = {}'.format(op, shape))
+    x = torch.randn((shape,), device='cuda', dtype=torch.float32)
+    z = torch.empty(
+        (1,),
+        device=x.device,
+        dtype=torch.float32,
+    )
+
+    kernel = patch_kernel(reduce1d_kernel, {'OP': op})
+    grid = (1,)
+    kernel[grid](x_ptr=x, z_ptr=z, block=shape)
+
+    if op == 'sum':
+        golden_z = torch.sum(x)
+    elif op == 'min':
+        golden_z = torch.min(x)
+    else:
+        golden_z = torch.max(x)
+
+    assert_close(z, golden_z, rtol=1e-3, atol=1e-3)
+
+
+reduce2d_configs = [
```

Comment:
I'd suggest also testing with different dtypes

Reply:
Unit tests for float16 / float32 / float64 added.

Integer types are blocked by other problem and will be added later.
https://github.com/openai/triton/blob/triton-mlir/lib/Conversion/TritonGPUToLLVM/TritonGPUToLLVM.cpp#L862